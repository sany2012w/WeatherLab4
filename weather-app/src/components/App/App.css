.App {
  width: 70%;
  margin: 2% 15%;
  font-family: sans-serif;
  background-color: rgba(197, 204, 213, 0.5);
  background-size: cover;
  background-repeat: no-repeat;
  padding: 1%;
  border-style: ridge;
  border-color: #3d485b;
  border-width: 3px;
}
