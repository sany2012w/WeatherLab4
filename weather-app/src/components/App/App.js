import React from 'react';
import './App.css';
import Geolocation from '../Geolocation/Geolocation';
import AddedCities from "../AddedCities/AddedCities";

function App() {
  return (
    <div className="App">
      <Geolocation />
      <AddedCities />
    </div>
  );
}

export default App;
