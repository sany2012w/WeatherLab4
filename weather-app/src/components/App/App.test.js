import React from "react";
import App from "./App";
import renderer from "react-test-renderer";
import configureMockStore from 'redux-mock-store';
import { Provider } from 'react-redux'
import thunk from "redux-thunk";
import weatherTestData from "./weatherTestData";

const testStore = configureMockStore([thunk]);

it(`App displayed correctly`, () => {
  let testCities = new Map([["London", weatherTestData], ["London1", weatherTestData]]);
  const store = testStore({
    geolocation: {
          forecast: weatherTestData,
          isLoading: false
      },
    cities: {
          added_cities: testCities
      }
    });
  const tree = renderer.create(
    <Provider store={store}>
      <App />
    </Provider>).toJSON();
  expect(tree).toMatchSnapshot();
});
