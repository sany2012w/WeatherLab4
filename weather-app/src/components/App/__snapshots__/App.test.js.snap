// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App displayed correctly 1`] = `
<div
  className="App"
>
  <div>
    <div
      className="header"
    >
      <div
        className="headerText"
      >
        Weather here
      </div>
      <button
        className="headerButton"
        onClick={[Function]}
      >
        Update Geolocation
      </button>
    </div>
    <div
      className="WeatherBlock"
    >
      <div
        className="City"
      >
        London
      </div>
      <div
        className="leftWeather"
      >
        <img
          className="Icon"
          src="https://openweathermap.org/img/wn/09d.png"
        />
        <div
          className="Temp"
        >
          7
           °C
        </div>
      </div>
      <div
        className="rightWeather"
      >
        <div
          className="Pressure"
        >
          Pressure: 
          1012
           hPa
        </div>
        <div
          className="Humidity"
        >
          Humidity: 
          81
          %
        </div>
        <div
          className="Clouds"
        >
          Clouds: 
          light intensity drizzle
        </div>
        <div
          className="Wind"
        >
          Wind: 
          4.1
           m/s
        </div>
        <div
          className="Coords"
        >
          Coordinates: [
          -0.13
          , 
          51.51
          ]
        </div>
      </div>
    </div>
  </div>
  <div>
    <div
      className="CitiesPanel"
    >
      <div
        className="CitiesText"
      >
        Favorites
      </div>
      <form
        onSubmit={[Function]}
      >
        <input
          className="CitiesInput"
          name="cityName"
          placeholder="City name"
          required={true}
          type="text"
        />
        <input
          className="CitiesButton"
          type="submit"
          value="Add city"
        />
      </form>
    </div>
    <div
      className="weather"
    >
      <div
        className="WeatherBlock"
      >
        <div
          className="City"
        >
          London
        </div>
        <div
          className="leftWeather"
        >
          <img
            className="Icon"
            src="https://openweathermap.org/img/wn/09d.png"
          />
          <div
            className="Temp"
          >
            7
             °C
          </div>
          <button
            className="button"
            onClick={[Function]}
          >
            Delete
          </button>
        </div>
        <div
          className="rightWeather"
        >
          <div
            className="Pressure"
          >
            Pressure: 
            1012
             hPa
          </div>
          <div
            className="Humidity"
          >
            Humidity: 
            81
            %
          </div>
          <div
            className="Clouds"
          >
            Clouds: 
            light intensity drizzle
          </div>
          <div
            className="Wind"
          >
            Wind: 
            4.1
             m/s
          </div>
          <div
            className="Coords"
          >
            Coordinates: [
            -0.13
            , 
            51.51
            ]
          </div>
        </div>
      </div>
      <div
        className="WeatherBlock"
      >
        <div
          className="City"
        >
          London
        </div>
        <div
          className="leftWeather"
        >
          <img
            className="Icon"
            src="https://openweathermap.org/img/wn/09d.png"
          />
          <div
            className="Temp"
          >
            7
             °C
          </div>
          <button
            className="button"
            onClick={[Function]}
          >
            Delete
          </button>
        </div>
        <div
          className="rightWeather"
        >
          <div
            className="Pressure"
          >
            Pressure: 
            1012
             hPa
          </div>
          <div
            className="Humidity"
          >
            Humidity: 
            81
            %
          </div>
          <div
            className="Clouds"
          >
            Clouds: 
            light intensity drizzle
          </div>
          <div
            className="Wind"
          >
            Wind: 
            4.1
             m/s
          </div>
          <div
            className="Coords"
          >
            Coordinates: [
            -0.13
            , 
            51.51
            ]
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
